nixos/modules/installer/tools/nixos-install.sh

```diff
@@ -86,7 +86,7 @@ if [[ ! -e $NIXOS_CONFIG && -z $system ]]; then
 fi
 
 # A place to drop temporary stuff.
-trap "rm -rf $tmpdir" EXIT
+trap 'rm -rf $tmpdir' EXIT
```

Comment:
Manually verified that the substitution here still works, even with single quotes (bash is hard). :+1: